<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1143</width>
    <height>702</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QWidget" name="OpenGL" native="true">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>0</y>
      <width>821</width>
      <height>641</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="ViewPosition">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>190</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>View Position</string>
    </property>
   </widget>
   <widget class="QPushButton" name="LineRotatioin">
    <property name="geometry">
     <rect>
      <x>990</x>
      <y>480</y>
      <width>111</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Line Rotate</string>
    </property>
   </widget>
   <widget class="QPushButton" name="DefaultPosition">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>570</y>
      <width>241</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Reset View</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="angle">
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>430</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="u">
    <property name="geometry">
     <rect>
      <x>870</x>
      <y>430</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="updatePosition">
    <property name="geometry">
     <rect>
      <x>1040</x>
      <y>420</y>
      <width>61</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Update</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>870</x>
      <y>60</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_15">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>410</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>angle</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_16">
    <property name="geometry">
     <rect>
      <x>870</x>
      <y>410</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>u</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>30</y>
      <width>241</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>Eye Coordinates</string>
    </property>
    <widget class="QDoubleSpinBox" name="eyeX">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-15.000000000000000</double>
     </property>
     <property name="value">
      <double>0.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>47</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>eye.X</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="eyeY">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>40</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-14.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>20</y>
       <width>47</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>eye.Y</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="eyeZ">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>40</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-16.000000000000000</double>
     </property>
     <property name="value">
      <double>0.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>20</y>
       <width>47</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>eye.Z</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>110</y>
      <width>241</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>Point Coordinates</string>
    </property>
    <widget class="QDoubleSpinBox" name="positionX">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-20.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>47</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>point.X</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="positionY">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>40</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-16.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>20</y>
       <width>47</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>point.Y</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>20</y>
       <width>47</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>point.Z</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="positionZ">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>40</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-13.000000000000000</double>
     </property>
     <property name="value">
      <double>0.000000000000000</double>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>250</y>
      <width>241</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>Line Start Coordinates</string>
    </property>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>47</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>b.X</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="bX">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>20</y>
       <width>47</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>b.Y</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="bY">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>40</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>20</y>
       <width>47</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>b.Z</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="bZ">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>40</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>330</y>
      <width>241</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>Line End Coordinates</string>
    </property>
    <widget class="QLabel" name="label_12">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>47</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>d.X</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="dX">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="dY">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>40</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="dZ">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>40</y>
       <width>62</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_13">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>20</y>
       <width>47</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>d.Y</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_14">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>20</y>
       <width>47</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>d.Z</string>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>610</y>
      <width>251</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Ensure you update the line before you rotate it</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1143</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>eyeX</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>892</x>
     <y>139</y>
    </hint>
    <hint type="destinationlabel">
     <x>909</x>
     <y>105</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>eyeY</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>959</x>
     <y>142</y>
    </hint>
    <hint type="destinationlabel">
     <x>989</x>
     <y>103</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>eyeZ</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1036</x>
     <y>141</y>
    </hint>
    <hint type="destinationlabel">
     <x>1059</x>
     <y>103</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>positionX</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>910</x>
     <y>189</y>
    </hint>
    <hint type="destinationlabel">
     <x>914</x>
     <y>206</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>positionY</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>977</x>
     <y>184</y>
    </hint>
    <hint type="destinationlabel">
     <x>980</x>
     <y>206</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>positionZ</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1028</x>
     <y>186</y>
    </hint>
    <hint type="destinationlabel">
     <x>1053</x>
     <y>201</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ViewPosition</sender>
   <signal>pressed()</signal>
   <receiver>MainWindow</receiver>
   <slot>viewPosition()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1017</x>
     <y>226</y>
    </hint>
    <hint type="destinationlabel">
     <x>1073</x>
     <y>259</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dX</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>903</x>
     <y>381</y>
    </hint>
    <hint type="destinationlabel">
     <x>924</x>
     <y>348</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dY</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>977</x>
     <y>384</y>
    </hint>
    <hint type="destinationlabel">
     <x>994</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dZ</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1026</x>
     <y>384</y>
    </hint>
    <hint type="destinationlabel">
     <x>1070</x>
     <y>355</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>angle</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1023</x>
     <y>437</y>
    </hint>
    <hint type="destinationlabel">
     <x>933</x>
     <y>403</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LineRotatioin</sender>
   <signal>pressed()</signal>
   <receiver>MainWindow</receiver>
   <slot>lineRotation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1032</x>
     <y>464</y>
    </hint>
    <hint type="destinationlabel">
     <x>1104</x>
     <y>436</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>DefaultPosition</sender>
   <signal>pressed()</signal>
   <receiver>MainWindow</receiver>
   <slot>defaultPosition()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1021</x>
     <y>517</y>
    </hint>
    <hint type="destinationlabel">
     <x>1099</x>
     <y>510</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>updatePosition</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>setLine()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>930</x>
     <y>458</y>
    </hint>
    <hint type="destinationlabel">
     <x>867</x>
     <y>463</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bZ</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1039</x>
     <y>332</y>
    </hint>
    <hint type="destinationlabel">
     <x>1052</x>
     <y>297</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bY</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>963</x>
     <y>333</y>
    </hint>
    <hint type="destinationlabel">
     <x>980</x>
     <y>292</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bX</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>889</x>
     <y>332</y>
    </hint>
    <hint type="destinationlabel">
     <x>903</x>
     <y>291</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>viewPosition()</slot>
  <slot>lineRotation()</slot>
  <slot>defaultPosition()</slot>
  <slot>toggleAnimation()</slot>
  <slot>outputs()</slot>
  <slot>updatePosition()</slot>
  <slot>setLine()</slot>
 </slots>
</ui>
